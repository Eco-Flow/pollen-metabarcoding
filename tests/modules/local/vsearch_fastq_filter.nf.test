nextflow_process {

    name "Test Process VSEARCH_FASTQ_FILTER"
    script "modules/local/vsearch_fastq_filter.nf"
    process "VSEARCH_FASTQ_FILTER"

    test("Should run without failures") {

        when {
            params {
                clean = false
            }
            process {
                """
                input[0] = Channel.of( [ [ "id":"ERR2537816", "single_end": false ], file("${projectDir}/data/filter-input/ERR2537816.assembled.fastq.gz") ] )
                """
            }
        }

        then {
            assert process.success
            with(process.out) {
              assert fasta.size() == 1
              assert fasta[0].size() == 2
              assert fasta[0][0].size() == 2
              assert path(fasta[0].get(1)).md5 == '6d9fa0fae486b91ca66f8ac12918de3a'
            }

        }

    }

}
