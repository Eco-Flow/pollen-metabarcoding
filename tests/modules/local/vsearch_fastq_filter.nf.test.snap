{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "ERR2537816",
                            "single_end": false
                        },
                        "ERR2537816.filtered.fasta:md5,6d9fa0fae486b91ca66f8ac12918de3a"
                    ]
                ],
                "1": [
                    "ERR2537816.filtered.log:md5,27e617273571932266a13bfcc41bbe84"
                ],
                "2": [
                    "versions.yml:md5,e12363d0bb077a8ace7181257a31a081"
                ],
                "fasta": [
                    [
                        {
                            "id": "ERR2537816",
                            "single_end": false
                        },
                        "ERR2537816.filtered.fasta:md5,6d9fa0fae486b91ca66f8ac12918de3a"
                    ]
                ],
                "log": [
                    "ERR2537816.filtered.log:md5,27e617273571932266a13bfcc41bbe84"
                ],
                "versions": [
                    "versions.yml:md5,e12363d0bb077a8ace7181257a31a081"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-03-18T14:36:15.512404"
    }
}